src/parser/DASHTree.cpp

```diff
@@ -1648,6 +1648,10 @@ bool DASHTree::open(const std::string& url, std::map<std::string, std::string> a
   }
 
   current_period_ = periods_[0];
+  if (has_timeshift_buffer_)
+  {
+    current_period_ = periods_.back();
+  }
```

Comment:
as done above one variable assignment replace the other one,
so this can be improved
```suggestion
  if (has_timeshift_buffer_)
    current_period_ = periods_.back();
  else
    current_period_ = periods_.front();
```

Reply:
or as alternative

`current_period_ = has_timeshift_buffer_ ? periods_.back() : periods_.front();`